<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="/morus"/>
  <arg name="rate" default="100"/>
  <!--arg name="attitude_ctrl_flag" default="true"/-->
  
  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find morus_control)/config/morus_control.yaml" command="load"/>

  <!--node pkg="morus_control" type="mav_state_pub.py" name="mav_state_pub" output="screen"  ns="$(arg namespace)">
  	<remap from="imu" to="ahrs" />
  </node-->

  <node pkg="morus_control" type="height_ctl.py" name="height_ctl" output="screen"  ns="$(arg namespace)">
  </node>
  
  <node pkg="morus_control" type="attitude_ctl.py" name="attitude_ctl" output="screen"  ns="$(arg namespace)">
  </node>

  <node pkg="morus_control" type="rotors_variation_controller_outputs_to_motor_velocities.py" name="rotors_variation_controller_outputs_to_motor_velocities" output="screen"  ns="$(arg namespace)">
  </node>

  <node pkg="morus_control" type="horizontal_ctl.py" name="horizontal_ctl" output="screen"  ns="$(arg namespace)">
  </node>

  <!--node pkg="morus_control" type="float2gmStatus.py" name="translator" ns="$(arg namespace)"/-->
  
  <!--group if="$(arg attitude_ctrl_flag)"-->
  <!-- If attitude_ctrl_flag is true run attitude mass controller -->
    <!-- load the controllers -->
    <!--  angle_tilt_0_controller  -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
      output="screen" ns="$(arg namespace)"  args="joint_state_controller
		  angle_wing_0_controller
		  angle_wing_1_controller 
		  angle_wing_2_controller 
		  angle_wing_3_controller">
  
  <remap from="/robot_description" to="$(arg namespace)/robot_description"/>
  </node>
    
  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
      respawn="false" output="screen"  ns="$(arg namespace)">
      <remap from="/joint_states" to="$(arg namespace)/joint_states"/>
      <remap from="/robot_description" to="$(arg namespace)/robot_description"/>
  </node>
  
    
  <!--/group-->


</launch>

